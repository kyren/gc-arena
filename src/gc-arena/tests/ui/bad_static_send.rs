use std::rc::Rc;

use gc_arena::{
    static_send::{static_send, StaticSend},
    Gc, MutationContext, Rootable,
};

fn make_send<'gc>(mc: MutationContext<'gc, '_>) -> StaticSend<Gc<'gc, Rc<i32>>> {
    let ptr = Gc::new(mc, Rc::new(3));
    static_send::<Rootable!['a => Gc<'a, Rc<i32>>]>(ptr)
}

fn main() {}
