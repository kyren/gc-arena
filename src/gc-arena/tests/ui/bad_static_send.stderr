error[E0277]: `Rc<i32>` cannot be sent between threads safely
  --> tests/ui/bad_static_send.rs:10:19
   |
10 |     static_send::<Rootable!['a => Gc<'a, Rc<i32>>]>(ptr)
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Rc<i32>` cannot be sent between threads safely
   |
   = help: the trait `Send` is not implemented for `Rc<i32>`
   = note: required for `Gc<'_, Rc<i32>>` to implement `Send`
note: required by a bound in `static_send`
  --> src/static_send.rs
   |
   |     Root<'static, R>: Send,
   |                       ^^^^ required by this bound in `static_send`
